frontend/src/Project/EditControls/CellTypeControls/TrainingUI/EmbeddingPlot.js

```diff
@@ -17,8 +40,27 @@ function EmbeddingPlot({ embedding }) {
         ? 'rgba(33,150,243,0.15)'
         : `rgba(${color[0]},${color[1]},${color[2]},1)`
     );
-    widthMap = widthMap.map((color, i) => (selection.includes(i) ? 1.3 : 0.5));
   }
+  if (predUncertainties) {
+    uncertaintyColors = predUncertainties.map((uncertainty) =>
+      isNaN(uncertainty) ? 'rgb(230,230,230)' : uncertainty
+    );
+  }
+  // Cells that are selected will have thicker markers
+  const widthMap = [...colorMap].map((_, i) => (selection.includes(i) ? 1.3 : 0.5));
+  // Color scale for uncertainty mapping
+  const colorScale = [
+    ['0', 'rgb(49,54,149)'],
+    ['0.111111111111', 'rgb(69,117,180)'],
+    ['0.222222222222', 'rgb(116,173,209)'],
+    ['0.333333333333', 'rgb(171,217,233)'],
+    ['0.444444444444', 'rgb(224,243,248)'],
+    ['0.555555555556', 'rgb(254,224,144)'],
+    ['0.666666666667', 'rgb(253,174,97)'],
+    ['0.777777777778', 'rgb(244,109,67)'],
+    ['0.888888888889', 'rgb(215,48,39)'],
```

Comment:
The hard-coding of `N/9` as strings seems strange, but if the underlying machinery expects strings and there is no better way to do formatted strings in javascript (i.e. something like what `f"(1 / 9)"` would do in Python) then +1.

Reply:
It's how they did it in the plotly documentation but I think I can just change it to (1 / 9).toFixed(12) for a 12 decimal place string.